concordium-consensus/src/Concordium/Queries.hs

```diff
@@ -934,6 +934,44 @@ getNextUpdateSequenceNumbers = liftSkovQueryStateBHI query
         updates <- BS.getUpdates bs
         return $ updateQueuesNextSequenceNumbers $ UQ._pendingUpdates updates
 
+-- | Get the index of accounts with scheduled releases.
+getScheduledReleaseAccounts ::
+    forall finconf.
+    BlockHashInput ->
+    MVR finconf (BHIQueryResponse [AccountPending])
+getScheduledReleaseAccounts = liftSkovQueryStateBHI query
+  where
+    query bs = do
+        releases <- BS.getScheduledReleaseAccounts bs
+        let processAtTimestamp ts accs = ([AccountPending acc ts | acc <- Set.toList accs] ++)
+        return $ Map.foldrWithKey processAtTimestamp [] releases
+
+-- | Get the index of accounts with stake in cooldown.
+getCooldownAccounts ::
+    forall finconf.
+    BlockHashInput ->
+    MVR finconf (BHIQueryResponse [AccountPending])
+getCooldownAccounts = liftSkovQueryStateBHI query
+  where
+    query bs = do
+        cooldowns <- BS.getCooldownAccounts bs
+        let processAtTimestamp ts accs = ([AccountPending acc ts | acc <- Set.toList accs] ++)
```

Comment:
Is the O(n) complexity of (++) a problem in this fold?

Reply:
It shouldn't be. The complexity is in the first argument of `(++)`, so it's important that we use `foldr`. When we consume this list produced by the fold, it is lazily constructed by consuming each set of accounts in turn.

Reply:
Maybe it's worth adding a comment here so that future reviewers of the code know that this has been taken into consideration.